### PR TITLE
feat: add an env to enable or disable block cache of an app

### PR DESCRIPTION
**Pull request by ZhongChaoqiang**
### What problem does this PR solve? 
Sometimes, there are some apps of different business in a cluster.Some apps  want to enable block cache, others my be not.
So we need add an env to enable or disable block cache of an app.

##### Related changes
- Need to update the documentation
- Need to be included in the release note

### CODE REVIEW

**Comment by ZhongChaoqiang** [comment]
The issue is apache/incubator-pegasus#795

**Review by levy5307 — DISMISSED** [review]

File: src/server/pegasus_server_impl.cpp [hunk 1]
```diff
@@ -2558,6 +2558,27 @@ void pegasus_server_impl::update_rocksdb_iteration_threshold(
...
+{
+    bool cache_enabled = _data_cf_rd_opts.fill_cache;
+    auto find = envs.find(ROCKSDB_BLOCK_CACHE_ENABLED);
+    if (find != envs.end()) {
+        if (!dsn::buf2bool(find->second, cache_enabled)) {
+            derror_replica("{}={} is invalid.", find->first, find->second);
+            return;
+        }
+    }
+
```
Comment (hycdong) [inline src/server/pegasus_server_impl.cpp:2572]: If this `ROCKSDB_BLOCK_CACHE_ENABLED` is not found in app_envs,  `cache_enabled` should be set as default value, but in your implementation, this value won't be updated.
For example, this default value is `true`. Firstly, admin user update it as `false` through adding an envs. Then **this app_envs is deleted**, the value is expected to be `true`, not `false`.

File: src/server/pegasus_server_impl.cpp [hunk 2]
```diff
@@ -2558,6 +2558,28 @@ void pegasus_server_impl::update_rocksdb_iteration_threshold(
...
+    bool cache_enabled = true;
+    auto find = envs.find(ROCKSDB_BLOCK_CACHE_ENABLED);
+    if (find != envs.end()) {
+        if (!dsn::buf2bool(find->second, cache_enabled)) {
+            derror_replica("{}={} is invalid.", find->first, find->second);
+            return;
+        }
+    }
+
+    if (_data_cf_rd_opts.fill_cache != cache_enabled) {
```
Comment (levy5307) [inline src/server/pegasus_server_impl.cpp:2574]: I think you can remove the `if` statement:
```
        ddebug_replica("update app env[{}] from \"{}\" to \"{}\" succeed",
                       ROCKSDB_BLOCK_CACHE_ENABLED,
                       _data_cf_rd_opts.fill_cache,
                       cache_enabled);
        _data_cf_rd_opts.fill_cache = cache_enabled;
```

Comment (levy5307) [reply src/server/pegasus_server_impl.cpp:2574]: Yes, you are right. Remain unchanged

**Review by levy5307 — APPROVED** [review]

**Review by hycdong — APPROVED** [review]

**Review by acelyc111 — APPROVED** [review]

**Pull request merged**